embedded-io/src/lib.rs

```diff
@@ -401,6 +401,15 @@ pub trait Write: ErrorType {
     /// If you are using [`WriteReady`] to avoid blocking, you should not use this function.
     /// `WriteReady::write_ready()` returning true only guarantees the first call to `write()` will
     /// not block, so this function may still block in subsequent calls.
+    ///
+    /// Unlike [`Write::write`], the number of bytes written is not returned. However, in the case of
+    /// writing to an `&mut [u8]` its possible to calculate the number of bytes written by subtracting
+    /// the length of the slice after the write, from the initial length of the slice.
+    ///
+    /// ```rust,ignore
+    /// let start = buf.len();
```

Comment:
for this to work you need to explicitly ensure `buf` is a slice: `let buf: &mut [u8] = &mut [0u8; 256];` can you add that to the snippet? 

not adding it creates a bit of a footgun: `let mut buf = [0u8; 256]` will compile but not work because the `write!` macro creates a temporary slice, so `buf.len()` afterwards will refer to the array len (so, 256), not to the shortened slice len.